add Tool Parser

luukunn opened the pull request:
增加tool parser解析能力
1.通过--tool-parser指定解析器，可以解析FC内容
2.通过--tool-parser-plugin指定解析文件，可以注册解析器

luukunn committed:
add tool-parser

luukunn committed:
add tool-parser

luukunn committed:
add tool parser

luukunn committed:
add tool parser

luukunn committed:
resolve merge conflicts

luukunn committed:
fix

LiqinruiG commented on fastdeploy/entrypoints/openai/api_server.py:
```diff
@@ -117,6 +120,7 @@ async def lifespan(app: FastAPI):
         args.reasoning_parser,
         args.data_parallel_size,
         args.enable_logprob,
+        args.tool_call_parser,
```
离线接口也需要增加

LiqinruiG commented on fastdeploy/entrypoints/openai/api_server.py:
```diff
@@ -100,6 +101,8 @@ async def lifespan(app: FastAPI):
     if args.tokenizer is None:
         args.tokenizer = args.model
+    if args.tool_parser_plugin:
+        ToolParserManager.import_tool_parser(args.tool_parser_plugin)
```
这个不适合放在多进程启动的模块中

LiqinruiG commented on fastdeploy/entrypoints/openai/serving_chat.py:
```diff
@@ -232,15 +232,21 @@ async def chat_completion_stream_generator(
-                        tool_calls=output.get("tool_call_content", []),
+                        tool_calls=None,
                     )
+                    previous_num_tokens += len(output["token_ids"])
```
上面的 continue， 会影响计数

LiqinruiG commented on fastdeploy/input/ernie_processor.py:
```diff
@@ -61,9 +62,9 @@ def __init__(self, model_name_or_path, reasoning_parser_obj=None):
         self.eos_token_id_len = len(self.eos_token_ids)
         self.pad_token_id = self.get_pad_id()
         self.reasoning_parser = None
+        self.tool_parser = tool_parser_obj
```
改成 self.tool_parser_obj 吧

LiqinruiG commented on fastdeploy/entrypoints/openai/serving_chat.py:
```diff
@@ -232,15 +232,21 @@ async def chat_completion_stream_generator(
-                    delta_message = DeltaMessage(
+                    if self.engine_client.data_processor.tool_parser and not res["finished"]:
+                        tool_delta_message = output["tool_delta_message"]
+                        if tool_delta_message is None:
```
思考场景，不能 continue。

luukunn committed:
add offline

luukunn committed:
Merge branch 'develop' into tool-parser

luukunn committed:
add offline

luukunn committed:
fix

AuferGachet committed:
parsers:tool&reasoning

AuferGachet committed:
parsers:tool&reasoning

AuferGachet committed:
修改tool parser名称·

AuferGachet committed:
update

luukunn committed:
Merge branch 'develop' into tool-parser

luukunn committed:
Merge branch 'tool-parser' of https://github.com/luukunn/FastDeploy into tool-parser

AuferGachet committed:
fix reasoning-parser

luukunn committed:
add requirements

luukunn committed:
Merge branch 'tool-parser' of https://github.com/luukunn/FastDeploy into tool-parser

luukunn committed:
fix finish reason

luukunn committed:
fix

luukunn committed:
Merge branch 'develop' into tool-parser

AuferGachet committed:
fix reasoning-parser

AuferGachet committed:
修改
Merge branch 'tool-parser' of https://github.com/luukunn/FastDeploy into tool-parser

luukunn committed:
fix

luukunn committed:
Merge branch 'tool-parser' of https://github.com/luukunn/FastDeploy into tool-parser

luukunn committed:
fix

luukunn committed:
fix

luukunn committed:
fix

luukunn committed:
Merge branch 'develop' into tool-parser

luukunn committed:
fix

luukunn committed:
fix
